packages/discovery-react-components/src/components/DocumentPreview/components/PdfHighlight/utils/Highlighter.ts

```diff
@@ -180,10 +180,16 @@ export class Highlighter {
 
     const { textToPdfTextItemMappings, textToHtmlBboxMappings } = this;
     if (textToPdfTextItemMappings) {
-      items = doMapping(items, textToPdfTextItemMappings, this.textMappingsLayout);
+      const mappedItems = doMapping(items, textToPdfTextItemMappings, this.textMappingsLayout);
+      if (mappedItems.length > 0) {
+        return mappedItems;
+      }
```

Comment:
When the length of the result of `doMapping` is zero, it means that mapping fails to find bboxes on PDF using PDF text items. In the case, mapping from bboxes on HTML fields is expected to be used as a fallback.

But the previous code didn't do that. So, this is to fix the behavior. In the current code, when mapping returns bboxes, use them. Fall back to use bbox form HTML fields otherwise.